tests/thread/stress_create.py

```diff
@@ -15,7 +15,8 @@ def thread_entry(n):
         _thread.start_new_thread(thread_entry, (thread_num,))
         thread_num += 1
     except MemoryError:
-        pass
+        # let other threads run
+        time.sleep(0.05)
```

Comment:
This shouldn't be necessary.  I think the reason you originally needed it on your esp32 port was because in certain cases the VM would never release the GIL.  This could happen if there was a short, infinite loop that had a try/except block which always raised an exception.  This would reset the GIL divisor back to its maximum and therefore the counter would never reach zero and never release the GIL.  This is fixed in the latest master.